### DISCUSSION TITLE
Can't hover when crazyflie trying to fly

### DISCUSSION THREAD

**Discussion by Residualstress** [bitcraze/discussions#1426] [Q&A]
Hi, whenever I want to take off the drone, it can't stop until crash to the ceiling. This will happen both when controlled by joystick with hover assist mode and command based flight control. I have checked the follow points:

- The crazyflie bolt firmware is latest.
- The flow deck works well. I lifted the drone with my hands, and its position estimates were reasonable.
- No error occurs in the console.
- Motors could spin when used `motorPowerSet` and `health.startPropTest`.
- I tied to increase the throttle very slightly through the joystick and the drone could take off and had a stable attitude. But I couldn't get it to hover, meaning that it would go up to the ceiling if I increased the throttle a little more.
- The environment is suitable for flying since I have done the crazyflie fly test many times, even autonomous tasks. The drone performed well before.

**This situation happened after a crash caused by my mistake. When I rebooted the drone, the ESC beep,  similar to the ECS are calibrating. But it didn't make any sound afterwards.**

Here are the console output, motor plot and position estimate of a failure take off example. In this example, I clicked  take off of command based flight control, and then clicked emergence stop when the drone nearly crashed the ceiling (around 2 meters from the ground) . Finally it fell down and I caught it with the net. 
```
SYS: ----------------------------
SYS: Crazyflie Bolt is up and running!
SYS: Build 38:f61da11d54b6 (2023.11 +38) MODIFIED
SYS: I am 0x20373347344D500E0057003A and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_INFO: CONFIG_DECK_FORCE=cpxOverUART2 found
DECK_INFO: compile-time forced driver cpxOverUART2 added
DECK_CORE: 3 deck(s) found
DECK_CORE: Calling INIT on driver bcFlow2 for deck 0
ZR2: Z-down sensor [OK]
PMW: Motion chip id: 0x49:0xB6
DECK_CORE: Calling INIT on driver bcLoco for deck 1
DECK_CORE: Calling INIT on driver cpxOverUART2 for deck 2
CPX-EXT-ROUTER: Waiting for CPX External router initialization
CPX-EXT-ROUTER: CPX External router initialized, CPX_VERSION: 0
IMU: BMI088: Using SPI interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushless motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2024.2 (CB10)
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
DECK_CORE: Deck 0 test [OK].
DECK_CORE: Deck 1 test [OK].
DECK_CORE: Deck 2 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 5384 bytes
AppStop is now: 1 
STAB: Starting stabilizer loop
SUP: Can not fly
SUP: Ready to fly
DWM: Automatic mode: detected TWR
SUP: Can not fly
SUP: Locked, reboot required
```
The 3 decks are prototyping deck, flow deck and  loco position deck.  Actually, this problem will happen even just installed flow deck.
![image](https://github.com/bitcraze/discussions/assets/92587824/cda5079d-6954-45df-969d-48fb120b7856)
![image](https://github.com/bitcraze/discussions/assets/92587824/c083eb0d-dd18-4ff4-a491-70b1208ef717)
I did another test to take off the drone by app layer with height is 0.2 m, the same problem happened. Here is my code in app layer and position estimate records:
```
void appMain()
{
  static setpoint_t setpoint;

  vTaskDelay(M2T(3000));
  
  paramVarId_t idPositioningDeck = paramGetVarId("deck", "bcFlow2");


  DEBUG_PRINT("Waiting for activation ...\n");

  uint8_t positioningInit = paramGetUint(idPositioningDeck);

  if (positioningInit) {
    float height_land = 0.2;
    float velFront = 0;
    float velSide = 0;
      if (1) {
        DEBUG_PRINT("Waiting for the client ...\n");
      	vTaskDelay(M2T(5000));
      	DEBUG_PRINT("Taking off ...\n");
 	int k = 0; // Initialize counter
        while (k < 500) {
        	vTaskDelay(M2T(10));
        	setHoverSetpoint(&setpoint, velFront, velSide, height, 0);
        	commanderSetSetpoint(&setpoint, 3);
        	k++;
        }
 	DEBUG_PRINT("k is %d now\n", k);

        while (1) {
        	vTaskDelay(M2T(10));     
        	memset(&setpoint, 0, sizeof(setpoint_t));
        	commanderSetSetpoint(&setpoint, 3);
        }     
        
      }
  } else {
    DEBUG_PRINT("No flow deck installed ...\n");
  }
} 
```
![image](https://github.com/bitcraze/discussions/assets/92587824/0e67534c-ea49-4425-a814-db105bfcb0b3)
It seems that the position estimate is accurate and the drone knows where it is but can't control itself.
Should I calibrate the ESCs? I never calibrated the ESCs even when I first assembled it, but the drone flew fine then!

**Comment by tobbeanton** [comment]
I would start by checking for vibrations (unbalanced motors/props) but also at PID tuning. Especially the Hover thrust
![image](https://github.com/bitcraze/discussions/assets/6593186/31ae94f8-9d54-4b18-b79b-814938298fc5)
